### PR TITLE
Switch external database guide to install on EL8

### PR DESCRIPTION
**Pull request by ehelms**
* [x] I am familiar with the [contributing](CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.3/Katello 4.5
* [ ] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

**Comment by ehelms** [comment]
@evgeni I broke the preview?

**Comment by ehelms** [comment]
I rebased on top of the fix for preview, but now I do not see anything that mentions it as a job? Does it have anything to do with:

```
[build-html](https://github.com/theforeman/foreman-documentation/runs/7114307046?check_suite_focus=true#step:11:14)
No files were found with the provided path: pr/. No artifacts will be uploaded.
```

**Comment by evgeni** [comment]
yeah…

https://github.com/theforeman/foreman-documentation/pull/1382

https://github.com/theforeman/foreman-documentation/pull/1384

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 1]
```diff
@@ -2,7 +2,7 @@
 = Installing PostgreSQL
 
 You can install only the same version of PostgreSQL that is installed with the `{foreman-installer}` tool during an internal database installation.
-You can install PostgreSQL using {RHELServer} 7 repositories or from an external source, as long as the version is supported.
+You can install PostgreSQL using {RHELServer} 8 repositories or from an external source, as long as the version is supported.
```
Comment (evgeni) [inline guides/common/modules/proc_installing-postgresql.adoc:5]: There is no RHEL 8 Server, only Zuul.

```suggestion
You can install PostgreSQL using {RHEL} 8 repositories or from an external source, as long as the version is supported.
```

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 2]
```diff
@@ -24,8 +19,8 @@ ifndef::orcharhino[]
 [options="nowrap" subs="+quotes,attributes"]
 ----
 # subscription-manager repos --disable '*'
-# subscription-manager repos --enable={RepoRHEL7ServerSoftwareCollections} \
---enable={RepoRHEL7Server} --enable={RepoRHEL7ServerSatelliteServerProductVersion}
+# subscription-manager repos --enable={RepoRHEL8Server} \
```
Comment (evgeni) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:22]:
```suggestion
# subscription-manager repos --enable={RepoRHEL8BaseOS} \
```
```suggestion
# subscription-manager repos --enable={RepoRHEL8Server} \
```

I wish asciidoc would fail on unknown attributes…

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 3]
```diff
@@ -2,15 +2,10 @@
 = Preparing a Host for External Databases
 
 ifndef::orcharhino[]
-Install a freshly provisioned system with the latest {RHEL} 7 server to host the external databases.
+Install a freshly provisioned system with the latest {RHEL} 8 server to host the external databases.
```
Comment (evgeni) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:5]:
```suggestion
Install a freshly provisioned system with the latest {RHEL} 8 to host the external databases.
```

There is no 8 server, only zuul

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc (hunk 2 above)
Comment (ehelms) [reply guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:22]: Just me your muse (:smile: https://github.com/theforeman/foreman-documentation/pull/1386/commits/5c3a270151fbe3c023d0f94765c8a902e3cfb3e1)

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 4]
```diff
@@ -2,7 +2,7 @@
 = Installing PostgreSQL
 
 You can install only the same version of PostgreSQL that is installed with the `{foreman-installer}` tool during an internal database installation.
-You can install PostgreSQL using {RHELServer} 7 repositories or from an external source, as long as the version is supported.
+You can install PostgreSQL using {RHEL} 8 repositories or from an external source, as long as the version is supported.
 {Project} supports PostgreSQL version 12.1.
```
Comment (evgeni) [inline guides/common/modules/proc_installing-postgresql.adoc:6]: Unrelated to this change, but we're currently deploying `postgresql-12.11` on EL8, and I guess should just document `12` here?

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 5]
```diff
@@ -2,15 +2,10 @@
...
-Install a freshly provisioned system with the latest {RHEL} 7 server to host the external databases.
+Install a freshly provisioned system with the latest {RHEL} 8 to host the external databases.
 endif::[]
 ifdef::orcharhino[]
-Install a freshly provisioned system with the latest CentOS 7, Oracle Linux 7, or {RHEL} 7 to host the external databases.
-endif::[]
-
-ifdef::satellite[]
-Subscriptions for Red{nbsp}Hat Software Collections and {RHEL} do not provide the correct service level agreement for using {Project} with external databases.
-You must also attach a {Project} subscription to the base operating system that you want to use for the external databases.
```
Comment (evgeni) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:13]: Why did you remove this block? We need the Satellite sub to get to the Satellite repo to get the `postgresql-evr` package.

I have no idea if the rest of the blob is true too (service level, I doubt it), but the repo/sub is important.

**Comment by evgeni** [comment]
I guess this should be picked to 3.1, 3.2 and 3.3?

File: guides/common/modules/proc_installing-postgresql.adoc (hunk 4 above)
Comment (ekohl) [reply guides/common/modules/proc_installing-postgresql.adoc:6]: Indeed, PostgreSQL dropped the concept of supported x.y releases with 10. Now they just support x.

**Comment by ehelms** [comment]
@mjivraja I have updated this if you'd like to also review.

**Comment by mjivraja** [comment]
> @mjivraja I have updated this if you'd like to also review.

The changes are pretty straightforward and there is no catch from my side. We can move forward by considering the cherry-pick to the Foreman versions 3.1 and later, as per the @evgeni 's latest comment.

**Review by evgeni — APPROVED** [review]
ACK on the content!

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 6]
```diff
@@ -2,8 +2,8 @@
 = Installing PostgreSQL
 
 You can install only the same version of PostgreSQL that is installed with the `{foreman-installer}` tool during an internal database installation.
-You can install PostgreSQL using {RHELServer} 7 repositories or from an external source, as long as the version is supported.
-{Project} supports PostgreSQL version 12.1.
+You can install PostgreSQL using {RHEL} 8 repositories or from an external source, as long as the version is supported.
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:5]:
```suggestion
You can install PostgreSQL using {EL} 8 repositories or from an external source, as long as the version is supported.
```

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 7]
```diff
@@ -2,20 +2,20 @@
 = Preparing a Host for External Databases
 
 ifndef::orcharhino[]
-Install a freshly provisioned system with the latest {EL} 7 server to host the external databases.
+Install a freshly provisioned system with the latest {EL} 8 server to host the external databases.
 endif::[]
 ifdef::orcharhino[]
-Install a freshly provisioned system with the latest CentOS 7, Oracle Linux 7, or {RHEL} 7 to host the external databases.
+Install a freshly provisioned system with the latest CentOS 8 Stream, Oracle Linux 8, or {RHEL} 8 to host the external databases.
```
Comment (ekohl) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:8]: @maximiliankolb can this also be simplified to `{EL} 8` and drop the whole ifdef or does Orcharhino want to explicitly mention some supported distros?

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 8]
```diff
@@ -25,12 +25,12 @@ ifndef::orcharhino[]
...
---enable={RepoRHEL7Server} --enable={RepoRHEL7ServerSatelliteServerProductVersion}
+# subscription-manager repos --enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream} --enable={RepoRHEL8ServerSatelliteServerProductVersion}
 endif::[]
 ifdef::foreman-el,katello[]
-# subscription-manager repos --enable={RepoRHEL7ServerSoftwareCollections} \
---enable={RepoRHEL7Server}
+# subscription-manager repos --enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream}
 endif::[]
```
Comment (ekohl) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:34]: I'd consider
```asciidoc
# subscription-manager repos \
ifdef::satellite[]
--enable={RepoRHEL8ServerSatelliteServerProductVersion} \
endif::[]
--enable={RepoRHEL8BaseOS} --enable={RepoRHEL8AppStream}

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 9]
```diff
@@ -2,20 +2,20 @@
...
+* The prepared host must meet {Project}'s {InstallingServerDocURL}storage-requirements_{project-context}[Storage Requirements].
+
 ifdef::satellite[]
-Subscriptions for Red{nbsp}Hat Software Collections and {RHEL} do not provide the correct service level agreement for using {Project} with external databases.
+Subscriptions for {RHEL} do not provide the correct service level agreement for using {Project} with external databases.
 You must also attach a {Project} subscription to the base operating system that you want to use for the external databases.
 endif::[]
 
-.Prerequisites
-* The prepared host must meet {Project}'s {InstallingServerDocURL}storage-requirements_{project-context}[Storage Requirements].
```
Comment (ekohl) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:17]: There are no more storage requirements?

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc (hunk 7 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:8]: EL8 is good for me for now! So yes, please drop the whole ifdef if that's the only diff.

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc (hunk 8 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:34]: Maybe also consider one parameter per line :)

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc (hunk 9 above)
Comment (ehelms) [reply guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:17]: Not on this line there isn't! See about 5 lines up. I think these got swapped around when I had to put something back. This will get adjusted in the next push.

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 10]
```diff
@@ -37,7 +36,7 @@ The base recommended external database configuration adjustments are as follows:
 * work_mem: 4MB
 
 ----
-# vi /var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf
+# vi /var/lib/pgsql/data/postgresql.conf
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:39]: Below this it doesn't render well. I think the block is missing `options="nowrap" subs="verbatim,quotes"]`.

Comment (maximiliankolb) [reply guides/common/modules/proc_installing-postgresql.adoc:39]: If you're using attributes, use `[options="nowrap", subs="+quotes,verbatim,attributes"]`.

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 11]
```diff
@@ -1,15 +1,10 @@
 [id="preparing-a-host-for-external-databases_{context}"]
 = Preparing a Host for External Databases
 
-ifndef::orcharhino[]
-Install a freshly provisioned system with the latest {EL} 7 server to host the external databases.
-endif::[]
-ifdef::orcharhino[]
-Install a freshly provisioned system with the latest CentOS 7, Oracle Linux 7, or {RHEL} 7 to host the external databases.
-endif::[]
+Install a freshly provisioned system with the latest {EL} 8 server to host the external databases.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:4]:
```suggestion
Install a freshly provisioned system with the latest {EL} 8 to host the external databases.
```

**Review by ekohl — COMMENTED** [review]
I'm not sure why, but if you look at https://theforeman-foreman-documentation-preview-pr-1378.surge.sh/nightly/Installing_Server/index-katello.html#installing-postgresql_foreman you can see it still renders poorly. This is also broken in the current guide, but I'd like to see it fixed.

Perhaps we should fix it first and then rebase this on top of it to make cherry picks easier?

File: guides/common/modules/proc_installing-postgresql.adoc (hunk 10 above)
Comment (evgeni) [reply guides/common/modules/proc_installing-postgresql.adoc:40]: is there a way to configure asciidoc to *always* use these settings for code blocks?

Comment (maximiliankolb) [reply guides/common/modules/proc_installing-postgresql.adoc:40]: I don't know, but it would be a nice to have for sure!

**Comment by evgeni** [comment]
the rendering seems to be fallout from https://github.com/theforeman/foreman-documentation/pull/1203, and the following patch seems to fix it for me:

```diff
diff --git guides/common/modules/proc_installing-postgresql.adoc guides/common/modules/proc_installing-postgresql.adoc
index 522a8f6eb..bd6b473a8 100644
--- guides/common/modules/proc_installing-postgresql.adoc
+++ guides/common/modules/proc_installing-postgresql.adoc
@@ -27,8 +27,11 @@ rh-postgresql12-postgresql-evr
 . Edit the `/var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf` file:
 +
 [options="nowrap" subs="verbatim,quotes"]
-
-Note that the default configuration of external PostgreSQL needs to be adjusted to work with {Project}. 
+----
+# vi /var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf
+----
++
+Note that the default configuration of external PostgreSQL needs to be adjusted to work with {Project}.
 The base recommended external database configuration adjustments are as follows:
 
 * checkpoint_completion_target: 0.9
@@ -36,10 +39,6 @@ The base recommended external database configuration adjustments are as follows:
 * shared_buffers: 512MB
 * work_mem: 4MB
 
-----
-# vi /var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf
-----
-+
 . Remove the `#` and edit to listen to inbound connections:
 +
 [options="nowrap" subs="verbatim,quotes"]
```

**Comment by evgeni** [comment]
Given this rendering issue in in 3.2+, I'll open a separate PR so we can cleanly pick the change into the affected releases.

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 12]
```diff
@@ -24,13 +23,13 @@ rh-postgresql12-postgresql-evr
...
-. Edit the `/var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf` file:
+. Edit the `/var/lib/pgsql/data/postgresql.conf` file:
 +
 [options="nowrap" subs="verbatim,quotes"]
 ----
-# vi /var/opt/rh/rh-postgresql12/lib/pgsql/data/postgresql.conf
+# vi /var/lib/pgsql/data/postgresql.conf
 ----
-+
+
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:32]: It said it can't generate the preview. Perhaps this `+` should not have been removed?

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 13]
```diff
@@ -12,9 +12,8 @@ include::snip_firewalld.adoc[]
 . To install PostgreSQL, enter the following command:
 +
 ----
-# yum install rh-postgresql12-postgresql-server \
-rh-postgresql12-syspaths \
-rh-postgresql12-postgresql-evr
+# dnf module enable postgresql:12
+# dnf install postgresql-server postgresql-evr
```
[the remark this answers is not in the document]
Comment (evgeni) [reply guides/common/modules/proc_installing-postgresql.adoc:16]: Yes, good catch!

We modified the repo instructions in `guides/common/modules/proc_preparing-a-host-for-external-databases.adoc` so there should also be some module enablement there

**Comment by ehelms** [comment]
Is this ready then to go in?

**Comment by evgeni** [comment]
@ehelms I think we're missing the enablement of the module that contains postgresql-evr

**Comment by ekohl** [comment]
I think users would get stuck on https://github.com/theforeman/foreman-documentation/pull/1378#discussion_r952705375 and should be addressed.

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — COMMENTED** [review]
Content wise :+1: but the build is failing. ACK once it's green.

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 14]
```diff
@@ -24,16 +19,28 @@ ifndef::orcharhino[]
...
+----
+# dnf module enable satellite:el8
+----
 endif::[]
+ifdef::katello[]
+----
+# dnf module enable katello:el8
 ----
 endif::[]
+endif::[]
```
Comment (evgeni) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:43]: This endif is too much

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 15]
```diff
@@ -24,16 +19,28 @@ ifndef::orcharhino[]
...
-ifdef::foreman-el,katello[]
-# subscription-manager repos --enable={RepoRHEL7ServerSoftwareCollections} \
---enable={RepoRHEL7Server}
+--enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream}
+----
++
+. Enable the following modules:
++
+[options="nowrap" subs="+quotes,attributes"]
```
Comment (evgeni) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:32]: this needs to be right before the `----`

I think you wanted:
```
[options="nowrap" subs="+quotes,attributes"]
----
ifdef::satellite[]
# dnf module enable satellite:el8
endif::[]
ifdef::katello[]
# dnf module enable katello:el8
endif::[]
----
```

**Comment by ehelms** [comment]
Woo, I think it's working now?

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 16]
```diff
@@ -12,9 +12,7 @@ include::snip_firewalld.adoc[]
 . To install PostgreSQL, enter the following command:
 +
 ----
-# yum install rh-postgresql12-postgresql-server \
-rh-postgresql12-syspaths \
-rh-postgresql12-postgresql-evr
+# dnf install postgresql-server postgresql-evr
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:15]: If you want to fix this for non-katello:
```suggestion
ifdef::katello,satellite,orcharhino[]
# dnf install postgresql-server postgresql-evr
endif::[]
ifndef::katello,satellite,orcharhino[]
# dnf install postgresql-server
endif::[]
```

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 17]
```diff
@@ -24,13 +19,23 @@ ifndef::orcharhino[]
 [options="nowrap" subs="+quotes,attributes"]
 ----
 # subscription-manager repos --disable '*'
+# subscription-manager repos \
+ifdef::satellite[]
+--enable={RepoRHEL8ServerSatelliteServerProductVersion} \
+endif::[]
```
Comment (ekohl) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:25]: There is no equivalent instruction for enabling the katello repos for upstream. Something like this should be close:
```
ifdef::katello[]
:distribution: rhel
:distribution-major-version: 8
:package-manager: dnf
include::proc_configuring-repositories-el.adoc[]
endif::[]
```

The only downside is that it also installs puppet which is actually not needed here.

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 18]
```diff
@@ -2,8 +2,8 @@
 = Installing PostgreSQL
 
 You can install only the same version of PostgreSQL that is installed with the `{foreman-installer}` tool during an internal database installation.
-You can install PostgreSQL using {RHELServer} 7 repositories or from an external source, as long as the version is supported.
-{Project} supports PostgreSQL version 12.1.
+You can install PostgreSQL using {EL} 8 repositories or from an external source, as long as the version is supported.
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:5]: Given we have `postgresql-evr`, is an external source really a valid option?
```suggestion
You can install PostgreSQL using {EL} 8 repositories.
```

Though given we previously already gave instructions to set up repositories, perhaps it's better to drop it.
```suggestion
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Apologies for the scope creep.

This whole chapter is very Katello-focused. I'd propose we drop the whole assembly for `foreman-el`. In guides/doc-Administering_Project/master.adoc we already have `ifndef::foreman-deb[]`. I'd propose we make that `ifdef::katello,orcharhino,satellite[]`. See https://github.com/theforeman/foreman-documentation/pull/1598.

File: guides/common/modules/proc_installing-postgresql.adoc [hunk 19]
```diff
@@ -2,8 +2,7 @@
 = Installing PostgreSQL
 
 You can install only the same version of PostgreSQL that is installed with the `{foreman-installer}` tool during an internal database installation.
-You can install PostgreSQL using {RHELServer} 7 repositories or from an external source, as long as the version is supported.
-{Project} supports PostgreSQL version 12.1.
+{Project} supports PostgreSQL version 12.
```
Comment (ekohl) [inline guides/common/modules/proc_installing-postgresql.adoc:5]: There's another occurrence of `PostgreSQL version 12.1` in another file (under considerations)

File: guides/common/modules/proc_preparing-a-host-for-external-databases.adoc [hunk 20]
```diff
@@ -24,13 +19,32 @@ ifndef::orcharhino[]
...
+--enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream}
+----
+ifdef::katello[]
++
+. Install the `katello-repos-latest.rpm` package
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# dnf localinstall https://yum.theforeman.org/katello/{KatelloVersion}/katello/el8/x86_64/katello-repos-latest.rpm
```
Comment (ekohl) [inline guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:35]: You'll also need the foreman-release RPM. This is because the katello module depends on foreman

**Review by ekohl — APPROVED** [review]
Thanks!

**Comment by ekohl** [comment]
How far should this be cherry picked? I don't see anything right now, but back then 3.4 wasn't branched either.

**Comment by evgeni** [comment]
> How far should this be cherry picked? I don't see anything right now, but back then 3.4 wasn't branched either.

IMHO 3.1+

**Comment by mjivraja** [comment]
Hello @evgeni @ekohl 
Are we going to cherry-pick this PR to v3.1 and later?

**Comment by ekohl** [comment]
I at least pushed it to 3.2 and later:
   6d1a84e42..62d1789ec  3.2 -> 3.2
   cfe63556d..f2d48fbcb  3.3 -> 3.3
   0f5168139..a7ab3465d  3.4 -> 3.4

I'd be happy to do the same for 3.1, but I'm always a bit more cautious with that. Should we discuss that in a separate PR?

**Comment by mjivraja** [comment]
> I at least pushed it to 3.2 and later:
>    6d1a84e42..62d1789ec  3.2 -> 3.2
>    cfe63556d..f2d48fbcb  3.3 -> 3.3
>    0f5168139..a7ab3465d  3.4 -> 3.4
> 
> I'd be happy to do the same for 3.1, but I'm always a bit more cautious with that. Should we discuss that in a separate PR?

Hi @ekohl 

I have confirmed with Evgeni. There is a need for cherry-picking v3.1 as well. 👍

**Comment by maximiliankolb** [comment]
IMHO this needs to be reworked before getting cherry-picked to 3.1. For example, I have concerns about "guides/common/modules/con_migrating-from-internal-databases-to-external-databases.adoc" which we use in downstream that is currently EL7 only.

**Comment by mjivraja** [comment]
> IMHO this needs to be reworked before getting cherry-picked to 3.1. For example, I have concerns about "guides/common/modules/con_migrating-from-internal-databases-to-external-databases.adoc" which we use in downstream that is currently EL7 only.

Ohk.
I will wait for the new PR then.

**Comment by ekohl** [comment]
> I will wait for the new PR then.

It is also fine to submit a cherry pick PR yourself. That's probably the best way to move forward with it. If you have the time, please do so. There we can address @maximiliankolb's concerns.
